src/yukon/backend/src/devserv/mock_server.py

```diff
@@ -0,0 +1,423 @@
+            # Launched as a separate thread so to run in parallel with
+            # the SSE asynchronous calls.
+            event_scheduler_th = threading.Thread(
+                target=self.event_scheduler, args=(description,))
+            event_scheduler_th.daemon = True
+            event_scheduler_th.start()
+
+            @ self.app.route(self.api_prefix + '/eventSource')
+            async def sse_node_update() -> Tuple[AsyncGenerator[bytes, None], Dict[str, str]]:
+                """
+                Establishes a central co-routine that updates at a defined
+                rate and sends server generated events at each iteration.
+                The specific event to be sent depends on what the event
+                scheduler thread defines, according to the defined timestamps
+                on the session description.
+                """
+                async def send_event(rate: float) -> AsyncGenerator[bytes, None]:
+                    while True:
+                        await asyncio.sleep(1 / rate)
+                        yield self.event.encode()
+
+                return Response(send_event(30.0), mimetype="text/event-stream")
+
+    def event_scheduler(self, description: dict) -> None:
+        """
+        Schedule events according to the defined timestamps
+        in each event of the session description.
+        """
+        if 'events' in description:
+            for event in description['events']:
+                # Set the time when the session starts WRT to the start of the
+                # Python mock server
+                session_start = 0.0
+                if event['starts_in']:
+                    session_start = event['starts_in']
+
+                # Events assotiated to the nodes
+                if event['nodes']:
+                    for idx, node in enumerate(event['nodes']):
+                        node_event = event['nodes'][node]
+
+                        self.schedule_node_status_update_events(node, node_event)
+                        self.schedule_publishers_events(node, node_event)
+                        self.schedule_subscribers_events(node, node_event)
+
+        # Waits for a defined amount of seconds before starting the session
+        while True:
+            if ((time() - self._session_timer_start) >= session_start):
+                sys.stdout.write('\033[34mMock session started...\n\033[0m')
+                self.session_scheduler.run()
+                break
+
+    def schedule_node_status_update_events(self, node: int, node_event: dict) -> None:
+        """
+        Node status change event
+        """
+        if 'status' in node_event:
+            status_event = node_event['status']
+
+            data = {
+                "id": int(node),
+                "health": status_event['health'],
+                "active": 1
+            }
+
+            # node appears
+            if 'timestamp_start' in status_event and status_event['timestamp_start']:
```

Comment:
I'm being borderline too pedantic here but did you mean to use `dict.get()`? This pattern occurs often in your code.

Reply:
Let's keep it as is for clarity.